### PR TITLE
Adjust open id provider login

### PR DESCRIPTION
**Pull request by larissazech**
Use example client to test the whole flow and decrypt the id_token containing the chosen username.

Since this branch was forked from add_open_id_provider, it might be necessary to merge PR #41 first and to rebase this one on master afterwards.

corresponding issue: #34

### CODE REVIEW

File: api/boilerplate/templates/check/check.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<!doctype html>
```
Comment (nils-wisiol) [inline api/boilerplate/templates/check/check.html:1]: please fix indentation in this file (according to #30 )

File: api/boilerplate/views.py [hunk 2]
```diff
@@ -1,7 +1,15 @@
 from django.contrib.auth.models import User, Group
 from rest_framework import viewsets
 from .serializers import UserSerializer, GroupSerializer
-
```
Comment (nils-wisiol) [inline api/boilerplate/views.py]: Please keep two lines blank before the begin of `class UserViewSet` ([PEP8](https://www.python.org/dev/peps/pep-0008/))

File: api/boilerplate/views.py [hunk 3]
```diff
@@ -17,3 +25,50 @@ class GroupViewSet(viewsets.ModelViewSet):
     """
     queryset = Group.objects.all()
     serializer_class = GroupSerializer
+
+# This function is based on django.contrib.auth.views.login
```
Comment (nils-wisiol) [inline api/boilerplate/views.py]: For function comments, please use `"""` (cf [PEP257](https://www.python.org/dev/peps/pep-0257/))

The [current (1.11.4) version](https://github.com/django/django/blob/1.11.4/django/contrib/auth/views.py#L136) of `django.contrib.auth.views.login` is just a stub that appears to be depreciated and superseded by a [class-based view](https://github.com/django/django/blob/1.11.4/django/contrib/auth/views.py#L67). Please clarify which function your code is based on.

File: api/entrypoint.sh [hunk 4]
```diff
@@ -4,7 +4,7 @@
 host=db; port=3306; n=120; i=0; while ! (echo > /dev/tcp/$host/$port) 2> /dev/null; do [[ $i -eq $n ]] && >&2 echo "$host:$port not up after $n seconds, exiting" && exit 1; echo "waiting for $host:$port to come up"; sleep 1; i=$((i+1)); done
 
 # collect static files
-python manage.py collectstatic --clear --noinput || exit 1
```
Comment (nils-wisiol) [inline api/entrypoint.sh]: The `collectstatic` command is designed to move static files from their locations in the different Django modules into a common directory. Afaik, it is not best practice to include static files into the common directory yourself. Please move the static files into their [canonical location](https://docs.djangoproject.com/en/1.11/ref/contrib/staticfiles/).

Also, keep in mind that you're adding duplicate code here. Much of the static files added here was already added in earlier commits (cf #25 )

**Review by nils-wisiol — CHANGES_REQUESTED** [review]
Please make sure to include e2e for the functionality you add with this PR.

**Comment by nils-wisiol** [comment]
(I changed the PR base to make an 'early' review possible. This should nevertheless be merged to `master`, not into `add_open_id_provider`.)

**Comment by larissazech** [comment]
Committed minor code style changes. 
Still missing: e2e tests.
Regarding issue #62, we propose to make massive changes on this code base. Therefore, the pull request has to wait until those changes are done and e2e tests are implemented.

**Comment by nils-wisiol** [comment]
is this currently waiting for review?

**Comment by BenjaminKeller** [comment]
No, we have to made changes according to #62. Then, we will call for review.

File: api/boilerplate/views.py [hunk 5]
```diff
@@ -17,3 +27,50 @@ class GroupViewSet(viewsets.ModelViewSet):
...
+def open_id_auth(request):
+
+    redirect_to='https://eid.local/api/eIdService/init'
+    params = None
+    if request.method == "GET":
+        params = request.GET
+    elif request.method == "POST":
+        params = request.POST
+    
+    return HttpResponseRedirect(redirect_to + '?' + params.urlencode())
```
Comment (BenjaminKeller) [inline api/boilerplate/views.py:40]: I wouldn't do it that way. You are hardcoding the hostname. Change
```
redirect_to = '/api/eIdService/init'
```
This works and you don't have to specify the hostname.

File: api/boilerplate/views.py [hunk 6]
```diff
@@ -17,3 +27,50 @@ class GroupViewSet(viewsets.ModelViewSet):
...
     serializer_class = GroupSerializer
+    
+def open_id_auth(request):
+
+    redirect_to='https://eid.local/api/eIdService/init'
+    params = None
+    if request.method == "GET":
+        params = request.GET
+    elif request.method == "POST":
+        params = request.POST
```
Comment (BenjaminKeller) [inline api/boilerplate/views.py:38]: Does the openId Protocol not specify if the request has to be made by POST or GET?

**Review by BenjaminKeller — COMMENTED** [review]
@larissazech Could delete all stuff which isn't needed? (ClientChecker etc.) I added two comments.

File: api/boilerplate/views.py (hunk 6 above)
Comment (larissazech) [reply api/boilerplate/views.py:38]: > Authorization Servers MUST support the use of the HTTP GET and POST methods defined in RFC 2616 [RFC2616] at the Authorization Endpoint. Clients MAY use the HTTP GET or POST methods to send the Authorization Request to the Authorization Server. If using the HTTP GET method, the request parameters are serialized using URI Query String Serialization, per Section 13.1. If using the HTTP POST method, the request parameters are serialized using Form Serialization, per Section 13.2.

taken from section 3.1.2.1 of [The OpenID Connect Core 1.0 Spec](http://openid.net/specs/openid-connect-core-1_0.html#AuthRequest)

**Comment by BenjaminKeller** [comment]
OK, great!

Von: larissazech<mailto:notifications@github.com>
Gesendet: Mittwoch, 30. August 2017 08:32
An: swp-fu-eid/eid-fu-swp<mailto:eid-fu-swp@noreply.github.com>
Cc: BenjaminKeller<mailto:benjamin-keller@live.de>; Comment<mailto:comment@noreply.github.com>
Betreff: Re: [swp-fu-eid/eid-fu-swp] Adjust open id provider login (#48)


@larissazech commented on this pull request.

________________________________

In api/boilerplate/views.py<https://github.com/swp-fu-eid/eid-fu-swp/pull/48#discussion_r135980839>:

> @@ -17,3 +27,50 @@ class GroupViewSet(viewsets.ModelViewSet):
     """
     queryset = Group.objects.all()
     serializer_class = GroupSerializer
+
+def open_id_auth(request):
+
+    redirect_to='https://eid.local/api/eIdService/init'
+    params = None
+    if request.method == "GET":
+        params = request.GET
+    elif request.method == "POST":
+        params = request.POST


Authorization Servers MUST support the use of the HTTP GET and POST methods defined in RFC 2616 [RFC2616] at the Authorization Endpoint. Clients MAY use the HTTP GET or POST methods to send the Authorization Request to the Authorization Server. If using the HTTP GET method, the request parameters are serialized using URI Query String Serialization, per Section 13.1. If using the HTTP POST method, the request parameters are serialized using Form Serialization, per Section 13.2.

taken from section 3.1.2.1 of The OpenID Connect Core 1.0 Spec<http://openid.net/specs/openid-connect-core-1_0.html#AuthRequest>

File: api/boilerplate/views.py (hunk 5 above)
Comment (larissazech) [reply api/boilerplate/views.py:40]: Thank you very much for the review! Done and tested, works fine!

File: api/boilerplate/views.py (hunk 6 above)
Comment (BenjaminKeller) [reply api/boilerplate/views.py:38]: OK, great.

**Comment by auvin** [comment]
Referring to #41, this is closed.

**Pull request closed without merging**